backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/domain/model/Delivery.java

```diff
@@ -0,0 +1,132 @@
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.puris.backend.delivery.domain.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Pattern;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import org.eclipse.tractusx.puris.backend.common.domain.model.measurement.ItemUnitEnumeration;
+import org.eclipse.tractusx.puris.backend.common.util.PatternStore;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Material;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Partner;
+
+import java.util.Date;
+import java.util.Objects;
+import java.util.UUID;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@SuperBuilder
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+@Entity
+@ToString
+public abstract class Delivery {
+    @Id
+    @GeneratedValue
+    protected UUID uuid;
+
+    @ManyToOne()
+    @JoinColumn(name = "partner_uuid")
+    @ToString.Exclude
+    @NotNull
+    protected Partner partner;
+
+    @ManyToOne()
+    @JoinColumn(name = "material_ownMaterialNumber")
+    @ToString.Exclude
+    @NotNull
+    protected Material material;
+
+    private double quantity;
+    private ItemUnitEnumeration measurementUnit;
+
+    @Pattern(regexp = PatternStore.NON_EMPTY_NON_VERTICAL_WHITESPACE_STRING)
+    private String trackingNumber;
+
+    private IncotermEnumeration incoterm;
+
+    // Order Position Reference
+    @Pattern(regexp = PatternStore.NON_EMPTY_NON_VERTICAL_WHITESPACE_STRING)
+    private String supplierOrderNumber;
+    @Pattern(regexp = PatternStore.NON_EMPTY_NON_VERTICAL_WHITESPACE_STRING)
+    private String customerOrderNumber;
+    @Pattern(regexp = PatternStore.NON_EMPTY_NON_VERTICAL_WHITESPACE_STRING)
+    private String customerOrderPositionNumber;
+
+    // Transit Location
+    @Pattern(regexp = PatternStore.BPNS_STRING)
+    private String destinationBpns;
+    @Pattern(regexp = PatternStore.BPNA_STRING)
+    private String destinationBpna;
+    @Pattern(regexp = PatternStore.BPNS_STRING)
+    private String originBpns;
+    @Pattern(regexp = PatternStore.BPNA_STRING)
+    private String originBpna;
+
+    // Transit Event
+    private Date dateOfDeparture;
```

Comment:
This model is a flattened Samm definition. The SAMM definition allows differentiation of actual and estimated times. See eventType property in [html](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.delivery_information/2.0.0/gen/DeliveryInformation.html) version. You can download the raw file and walkthrough it.

Either you need an enumeration like eventType and add it for both dates or add additional dates. First option is more aligned to the samm.

Please remember to update the validations of the entities